docker-storage-setup.sh

```diff
@@ -657,26 +657,36 @@ EOF
     Fatal "udevadm settle after partition creation failed. Exiting."
   fi
 
-  if ! wait_for_dev ${dev}1; then
-    Fatal "Partition device ${dev}1 is not available"
+  if ! wait_for_dev ${part}; then
+    Fatal "Partition device ${part} is not available"
   fi
 }
 
+partition_name() {
+    if echo "$1" | grep -q '[0-9]$'; then
+        echo "${1}p1"
+    else
+        echo "${1}1"
+    fi
+}
+
 create_disk_partitions() {
-  local devs="$1"
+  local devs="$1" part
 
   for dev in $devs; do
-    create_partition $dev
+    part=$(partition_name $dev)
+
+    create_partition $dev $part
```

Comment:
create_partition already outputs to stdout, so we would have to make it stop doing that first...